### DISCUSSION TITLE
Allow not typing brackets in functions declaration if there are no args

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4701] [Scripting]
Ruby:
```ruby
def f
  puts 'hi'
end

f()
```

In GDScript it could be like
```gdscript
func f:
    print('hi')
```
Or even without `:` (See https://github.com/godotengine/godot-proposals/discussions/4699)
```gdscript
func f
    print('hi')
```

**Comment by dalexeev** [comment]
We prefer not to have multiple syntax options _doing the same thing_. For example, we have two dictionary styles, but the Lua style allows you to omit quotes for keys. We have two setter/getter styles, but the alternative style allows you to use a separate function like `setget` in 3.x. An alternative function declaration syntax would not bring any bonuses other than aesthetic ones. I think it's not worth it, it would only complicate parser, clutter documentation, etc.

Also, judging by the reactions and activity over almost 2 years, this proposal did not receive support.

However, we have 2 precedents for optional empty parentheses:

* `signal my_signal` vs `signal my_signal()`;
* `get:` vs `get():` (see #8057).

Of course, we can say that they exist for historical reasons and this is not a sufficient reason to introduce another exception. It is also unclear what benefits the optional empty parentheses provide. #8057 requested support for parentheses, not their optionality.

**Let's keep this discussion open for a while and see if the community shows interest in it, maybe new arguments will appear.**
